client/webserver/passbytes.go

```diff
@@ -0,0 +1,188 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package webserver
+
+import (
+	"fmt"
+	"strconv"
+	"unicode"
+	"unicode/utf16"
+	"unicode/utf8"
+)
+
+const (
+	escapeSequence        = '\\'
+	unicodePrefix         = 'u'
+	unicodeSequenceLength = 6
+)
+
+// PassBytes is an alias of type []byte that implements json.UnmarshalJSON
+// to enable parsing json-encoded strings as []byte.
+type PassBytes []byte
+
+// UnmarshalJSON satisfies the json.Unmarshaler interface, parses json-encoded
+// strings as []byte. Returns an error if the raw bytes to be unmarshalled is
+// not a valid json-encoded string.
+func (pb *PassBytes) UnmarshalJSON(rawBytes []byte) error {
+	rawBytes, err := sanitizeJSONEncodedString(rawBytes)
+	if err != nil {
+		return fmt.Errorf("cannot unmarshal password: %v", err)
+	}
+	*pb = make([]byte, len(rawBytes))
+	copy((*pb)[:], rawBytes)
+	return nil
+}
+
+func (pb PassBytes) String() string {
+	return string(pb)
+}
+
+// sanitizeJSONEncodedString receives a json-encoded string in []byte and
+// returns a new []byte representing the same string but without quotes and
+// escape characters.
+func sanitizeJSONEncodedString(stringBytes []byte) ([]byte, error) {
+	if stringBytes[0] == '"' {
+		if len(stringBytes) < 2 || stringBytes[len(stringBytes)-1] != '"' {
+			return nil, fmt.Errorf("data is not properly quoted")
+		}
+		// string is quoted, remove first and last bytes
+		stringBytes = stringBytes[1 : len(stringBytes)-1]
+	}
+
```

Comment:
And tested.

Reply:
This review helped clarify things up for me a little bit. Since when unmarshalling, we expect a proper json-encoded data, I did a little test (using `json.Valid([]byte)`) that demonstrate that a valid json-encoded data cannot be unquoted.

See https://play.golang.org/p/-mXpjcmyxUd.

I'll now add a first check for valid json-encoding, followed by a check for presence of quotes at the beginning and end of the data.